extensions/jdbc/src/main/java/io/deephaven/jdbc/JdbcTypeMapper.java

```diff
@@ -0,0 +1,1321 @@
+     * @param columnName the column for which to find a type match
+     * @param deephavenDataType if not null, the desired Deephaven target data type
+     * @param throwException if true, throw an exception if a mapping cannot be found. Otherwise, return null
+     * @param <T> the target Deephaven type
+     * @return a mapping object that can be used to map SQL/JDBC ResultSet values to and from Deephaven column values
+     */
+    static public <T> DataTypeMapping<T> getColumnTypeMapping(final Connection connection,
+            final ResultSetMetaData rs, final String columnName,
+            Class<T> deephavenDataType, boolean throwException) {
+        final int type;
+        Class<? extends Connection> connClass = null;
+        try {
+            type = rs.getColumnType(getMetaDataColumnIndex(rs, columnName));
+
+            // statement/connection should be null only in test cases (in which case we cannot test driver-specific
+            // behavior anyway)
+            if (connection != null) {
+                connClass = connection.getClass();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Failed to get metadata for source column: " + columnName, e);
+        }
+        return getColumnTypeMapping(connClass, type, deephavenDataType, throwException);
+    }
+
+    /**
+     * Get type mapping for converting JDBC to/from Deephaven column type. If the target Deephaven type is not
+     * specified, a default will be used.
+     *
+     * @param connClass a JDBC Connection subclass, used to determine database-specific mappings
+     * @param type the SQL type
+     * @param deephavenDataType if not null, the desired Deephaven target data type
+     * @param <T> the target Deephaven type
+     * @return a mapping object that can be used to map SQL/JDBC ResultSet values to and from Deephaven column values
+     */
+    static public <T> DataTypeMapping<T> getColumnTypeMapping(final Class<? extends Connection> connClass, int type,
+            Class<? extends T> deephavenDataType) {
+        return getColumnTypeMapping(connClass, type, deephavenDataType, false);
+    }
+
+    /**
+     * Get type mapping for converting JDBC to/from Deephaven column type. If the target Deephaven type is not
+     * specified, a default will be used.
+     *
+     * @param connClass a JDBC Connection subclass, used to determine database-specific mappings
+     * @param type the SQL type
+     * @param deephavenDataType the column type, if known. If null, the default type mapping will be provided
+     * @param throwException if true, throw an exception if a mapping cannot be found. Otherwise, return null
+     * @param <T> the target Deephaven type
+     * @return a mapping object that can be used to map SQL/JDBC ResultSet values to and from Deephaven column values
+     */
+    static public <T> DataTypeMapping<T> getColumnTypeMapping(final Class<? extends Connection> connClass, int type,
```

Comment:
You mentioned wanting to use the one that throw an exception. Return null on no exception should be specified on all of these.